concordium-consensus/src/Concordium/GlobalState/Persistent/BlockState.hs

```diff
@@ -3673,6 +3705,31 @@ migrateBlockPointers migration BlockStatePointers{..} = do
               bspRewardDetails = newRewardDetails
             }
 
+-- | Thaw the block state, making it ready for modification.
+--  This function wraps the underlying 'PersistentBlockState' of the provided 'HashedPersistentBlockState' in a new 'IORef'
+--  such that changes to the thawed block state does not propagate into the parent state.
+--
+--  Further the 'DiffMap.DifferenceMap' of the accounts structure in the provided block state is
+--  "bumped" in the sense that a new one is created for the new thawed block with a pointer to the parent difference map.
+--  The parent difference map is empty if the parent is persisted otherwise it may contain new accounts created in that block.
+doThawBlockState ::
+    (SupportsPersistentState pv m) =>
+    HashedPersistentBlockState pv ->
+    m (PersistentBlockState pv)
+doThawBlockState HashedPersistentBlockState{..} = do
+    -- This load is cheap as the underlying block state is retained in memory as we're building from it, so it must be the "best" block.
+    bsp@BlockStatePointers{bspAccounts = a0@Accounts.Accounts{..}} <- loadPBS hpbsPointers
+    mDiffMap <- liftIO $ readIORef accountDiffMapRef
+    newDiffMapRef <- case mDiffMap of
+        -- reuse the reference pointing to @Nothing@.
+        Absent -> return accountDiffMapRef
```

Comment:
This looks incorrect to me if we derive more than one block from the same parent. Won't this cause issues if those blocks have new accounts?

Reply:
Yes, this is incorrect. It's fixed. ( I still have to add a few tests.)